corehq/apps/sms/api.py

```diff
@@ -206,11 +212,44 @@ def send_message_via_backend(msg, backend=None, orig_phone_number=None):
         return False
 
 
+def random_password():
+    """
+    This method creates a random password for an sms user registered via sms
+    """
+    chars = string.ascii_uppercase + string.ascii_lowercase + string.digits
+    return ''.join(random.choice(chars) for x in range(15))
+
+
+def process_username(username, domain):
+    """
+    Loosely based on code from apps/users/forms.py:255
+    """
+    from corehq.apps.users.forms import validate_username
+
+    max_len_username = 80
+    if len(username) > max_len_username:
+        raise forms.ValidationError(
+            "Username %s is too long.  Must be under %d characters."
+            % (username, max_len_username))
+    # Check if the username contains invalid characters w/ django checker
+    validate_username('%s@commcarehq.org' % username)
+    username = format_username(username, domain)
+    num_couch_users = len(CouchUser.view("users/by_username",
+                                         key=username,
+                                         reduce=False))
+    if num_couch_users > 0:
+        # This should never occur, line 377 should check if v (verified number) exists already:
+        #     if v is not None and v.verified:
```

Comment:
I think we should remove this comment - just because a phone number doesn't belong to anyone doesn't mean necessarily that the requested username is available.